src/kernels/attention_kernels.cpp

```diff
@@ -155,13 +163,27 @@ void crossAttention(bfloat16_t *output, bfloat16_t *query, bfloat16_t *key, bflo
     int maxCtxSize = 0;
     int blkOffsets[batchSize]; // offset in blockTables
     int curOff = 0;
+
+    if (unlikely(threadNum == 0)) {
+#pragma omp parallel
+        {
+            int tid = omp_get_thread_num();
+            if (tid == 0) { threadNum = omp_get_num_threads(); }
+        }
+    }
+
+    // blocktables dim = 2
     for (int i = 0; i < batchSize; ++i) {
         if (contextSizes[i] > maxCtxSize) { maxCtxSize = contextSizes[i]; }
+    }
+
+    int max_block_num = (maxCtxSize + cacheBlkSize - 1) / cacheBlkSize;
```

Comment:
Could you pls make the name in camelCase? to align with the overall style.
@aurora327